server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -561,6 +561,44 @@ private IndexMetadata(
         assert numberOfShards * routingFactor == routingNumShards : routingNumShards + " must be a multiple of " + numberOfShards;
     }
 
+    IndexMetadata(IndexMetadata indexMetadata, MappingMetadata mapping) {
```

Comment:
Could we maybe make this just a method:

`IndexMetadata withMappingMetadata(MappingMetadata mapping)` to give us an obvious copy constructor. I also wonder, even though it's slightly more code, maybe it's better if this would just invoke the existing constructor so we don't have two constructors to maintain for this thing (given how additional fields are incoming from settings moving to fields in this class)?

Reply:
👍 I agree this is cleaner. I've made this change: 7d092821070b0983e4280a07be21251647e40636